foundry/mouseInteractionManager.d.ts

```diff
@@ -1,213 +1,257 @@
+      | 'dragLeftCancel'
+      | 'dragLeftDrop'
+      | 'dragLeftMove'
+      | 'dragLeftStart'
+      | 'dragRightCancel'
+      | 'dragRightDrop'
+      | 'dragRightMove'
+      | 'dragRightStart'
+      | 'hoverIn'
+      | 'hoverOut',
+      ((user: User, event: PIXI.InteractionEvent) => boolean) | boolean
+    >
+  >;
+  callbacks: Record<keyof this['permissions'], ((event: Event | PIXI.InteractionEvent) => any) | null>;
+  options: { target?: string[] | string | null };
 
   /**
-   * The drag handling time
+   * The current interaction state
+   * @defaultValue `0`
    */
-  dragTime: number;
+  state: ValueOf<typeof MouseInteractionManager['INTERACTION_STATES']>;
 
   /**
    * Bound handlers which can be added and removed
+   * @defaultValue `{}`
    */
-  handlers: { [x: string]: Function };
+  handlers: Record<
+    'contextmenu' | 'mousedown' | 'mousemove' | 'mouseout' | 'mouseover' | 'mouseup' | 'rightdown',
+    Function
+  >;
 
-  layer: PlaceablesLayer;
+  /**
+   * The drag handling time
+   * @defaultValue `0`
+   */
+  dragTime: number;
+
+  /**
+   * The throttling time below which a mouse move event will not be handled
+   * @defaultValue `Math.ceil(1000 / canvas.app.ticker.maxFPS)`
+   */
+  protected _dragThrottleMS: number;
 
   /**
    * The time of the last left-click event
+   * @defaultValue `0`
    */
   lcTime: number;
 
-  object: PlaceableObject;
-
-  options: object;
-
-  permissions: object;
-
   /**
    * The time of the last right-click event
+   * @defaultValue `0`
    */
   rcTime: number;
 
-  /**
-   * The current interaction state
-   */
-  state: number;
-
   /**
    * A flag for whether we are right-click dragging
+   * @defaultValue `false`
    */
   protected _dragRight: boolean;
 
-  constructor(
-    object: PlaceableObject,
-    layer: PlaceablesLayer,
-    permissions?: object,
-    callbacks?: object,
-    options?: object
-  );
-
-  /**
-   * A reference to the possible interaction states which can be observed
-   */
-  get states(): { [x: string]: number };
-
   /**
    * Get the target
+   * @returns `this.object` or `this.object[this.options.target]`
    */
-  get target(): any;
+  get target(): T;
 
   /**
    * Activate interactivity for the handled object
    */
-  activate(): MouseInteractionManager;
-
-  /**
-   * Execute a callback function associated with a certain action in the workflow
-   * @param action - The action being attempted
-   * @param event - The event being handled
-   */
-  callback(action: string, event: Event): any;
+  activate(): this;
 
   /**
    * Test whether the current user has permission to perform a step of the workflow
    * @param action - The action being attempted
-   * @param event - The event being handled
+   * @param event  - The event being handled
    * @returns Can the action be performed?
    */
-  can(action: string, event: Event): boolean;
+  can(action: keyof this['permissions'], event: Event | PIXI.InteractionEvent): boolean;
```

Comment:
`keyof this['permissions']` is used quite often, it might make sense to extract it as a type to improve readability.